### PR TITLE
system-variables: add init connect

### PR DESCRIPTION
**Pull request by morgo**
### What is changed, added or deleted? (Required)

This is a minor feature that is MySQL compatible.

The TiDB PR is: https://github.com/pingcap/tidb/pull/23713

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

https://github.com/pingcap/tidb/pull/23713

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by xhebox — APPROVED** [review]
LGTM

File: system-variables.md [hunk 1]
```diff
@@ -99,6 +99,14 @@ mysql> SELECT * FROM t1;
 - Default value: (system hostname)
 - The hostname of the TiDB server as a read-only variable.
 
+### init_connect
+
+- Scope: GLOBAL
+- Default value: ""
+- The `init_connect` feature permits a SQL statement to be executed when users first connect to a TiDB server.
+- Users that have the `CONNECTION_ADMIN` or `SUPER` privileges will not execute the `init_connect` statement.
+- If the `init_connect` statement results in an error, the user connection will be terminated.
```
Comment (TomShawn) [inline system-variables.md:108]:
```suggestion
- The `init_connect` feature permits a SQL statement to be automatically executed when you first connect to a TiDB server. If you have the `CONNECTION_ADMIN` or `SUPER` privileges, this `init_connect` statement will not be executed. If the `init_connect` statement results in an error, the user connection will be terminated.
```

Comment (qiancai) [reply system-variables.md:108]: @morgo, please check the comment from Tom as above and update the docs if applicable. Thanks!

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**